OpenSim/Sandbox/MatlabScripts/muscleCurves.m

```diff
@@ -0,0 +1,259 @@
+% R24 HD065690) and by DARPA through the Warrior Web program.             %
+%                                                                         %   
+% Copyright (c) 2005-2012 Stanford University and the Authors             %
+% Author(s): James Dunne                                                  %
+%                                                                         %
+% Licensed under the Apache License, Version 2.0 (the "License");         %
+% you may not use this file except in compliance with the License.        %
+% You may obtain a copy of the License at                                 %
+% http://www.apache.org/licenses/LICENSE-2.0.                             %
+%                                                                         % 
+% Unless required by applicable law or agreed to in writing, software     %
+% distributed under the License is distributed on an "AS IS" BASIS,       %
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         %
+% implied. See the License for the specific language governing            %
+% permissions and limitations under the License.                          %
+% ----------------------------------------------------------------------- %
+
+% Author: James Dunne, Chris Dembia, Tom Uchida, Ajay Seth.   
+   
+
+%%
+function [fl_active,fl_passive] = muscleCurves(modelpath)
+
+
+
+import org.opensim.modeling.*      % Import OpenSim Libraries
+
+
+if nargin < 1
+    [filein, pathname] = uigetfile({'*.osim','osim'}, 'OSIM model file...');
+    model = Model(fullfile(pathname,filein));
+elseif nargin == 1
+    model = Model(modelpath);
+end
+
+s =  model.initSystem();
+
+
+%% select a muscle
+musclelist = {};
+name = [];
+stoploop = [];
+
+for i = 0 : model.getMuscles.getSize() - 1
+    musclelist = [ musclelist ; {char(model.getMuscles.get(i).getName)} ];
+    display( char(model.getMuscles.get(i).getName ) ) 
+end
+ display('(type exit to stop).');
+
+
+
+while isempty(stoploop)
+
+    %%%
+    while isempty(name)
+        
+       
+        string = input('feed me a muscle name!?', 's');
+        
+        % get a valid name or 'exit' 
+        nameIndex = find( cellfun(@(s) ~isempty(strmatch(string, s)), musclelist) == 1);
+        
+        if strmatch(string, 'exit');
+            stoploop = 1;
+            break
+        elseif isempty(nameIndex);
+            display('Muscle name not found.');
+        else  
+            name = 1;
+            musclename = char( musclelist(nameIndex) );
+        end
+    end   
+    %%%     
+        
+    if ~isempty(stoploop)
+        display('thanks for ending this program. See you Laterz!')
+        close all
+        clc
+        return
+    end
+        
+    %% Muscle Coordinate finder
+    %   Find the coordinate's that each muscle crosses. This is done by
+    %   examining the moment arm contribution of the muscle across all
+    %   coordinates. A muscle will contribute to any coodinate when the moment
```

Comment:
I updated this section to check the moment arm at three locations within the range of the coordinate.